Libraries/LibWeb/HTML/Scripting/Environments.h

```diff
@@ -115,6 +118,8 @@ struct EnvironmentSettingsObject : public Environment {
     [[nodiscard]] bool discarded() const { return m_discarded; }
     void set_discarded(bool b) { m_discarded = b; }
 
+    virtual void run_environment_discarding_steps() override;
```

Comment:
Opinion: instead of literally adopting "run (the) environment discarding steps" as a name, we could call it "discard_environment" to improve readability without it being confusing. It's still something we can find by the anchor ref from the spec since a spec link is present in the implementation.

Reply:
That's fair! I know there was a discussion about this kind of naming but I didn't remember if we came to a conclusion. I didn't like this name but couldn't think of anything better.